security/i2pd/files/0002-util.cpp-do-not-use-pthread_setname_np-on-unsupporte.patch

```diff
@@ -0,0 +1,44 @@
+From 22295b6527e780a577d2756bb68eb4673e745a6c Mon Sep 17 00:00:00 2001
+From: Sergey Fedorov <vital.had@gmail.com>
+Date: Mon, 31 Oct 2022 12:17:53 +0800
+Subject: [PATCH 2/2] util.cpp: do not use pthread_setname_np on unsupported
+ macOS versions
+
+---
+ libi2pd/util.cpp | 12 +++++++++++-
+ 1 file changed, 11 insertions(+), 1 deletion(-)
+
+diff --git a/libi2pd/util.cpp b/libi2pd/util.cpp
+index 422c97ff..b8234b39 100644
+--- a/libi2pd/util.cpp
++++ b/libi2pd/util.cpp
+@@ -22,6 +22,9 @@
+ #include <pthread_np.h>
+ #endif
+ 
++#if defined(__APPLE__)
++# include <AvailabilityMacros.h>
++#endif
+ 
+ #ifdef _WIN32
+ #include <stdlib.h>
+@@ -143,8 +146,15 @@ namespace util
+ 	}
+ 
+ 	void SetThreadName (const char *name) {
+-#if defined(__APPLE__) && !defined(__POWERPC__)
++#if defined(__APPLE__)
++# if (!defined(MAC_OS_X_VERSION_10_6) || \
++		(MAC_OS_X_VERSION_MAX_ALLOWED < 1060) || \
++		defined(__POWERPC__))
++		/* pthread_setname_np is not there on <10.6 and all PPC.
```

Comment:
Why not use legacysupport instead?

Reply:
@catap I think it was tried and didn’t work. Also, that has been merged to upstream already. It is a temporary patch.

Reply:
@catap By the way, we can drop PPC patches now, updating to 2.44.0: https://github.com/PurpleI2P/i2pd/blob/openssl/ChangeLog